pkg/parser/additional/additional.go

```diff
@@ -0,0 +1,199 @@
+
+func expirationDateFromPemCertificate(filePath string) [3]int {
+	out, err := exec.Command("openssl", "x509", "-in", filePath, "-noout", "-dates").Output()
+	if err != nil {
+		log.Error().Msgf("Failed to execute command 'openssl x509 -in %s -noout -dates'", filePath)
+	}
+
+	date := getExpirationDate(string(out))
+
+	return date
+}
+
+func getBitRSAKeyBit(filePath string) int {
+	out, err := exec.Command("openssl", "x509", "-in", filePath, "-text", "-noout").Output()
+
+	if err != nil {
+		log.Error().Msgf("Failed to execute command 'openssl x509 -in %s -text -noout'", filePath)
+		return -1
+	}
+
+	rsaInfo := getRegexMatch(`RSA Public-Key:[ ]\([0-9]{4}[ ]bit\)`, string(out))
+
+	bits := getRegexMatch(`[0-9]{4}`, rsaInfo)
+	rsabits, err := strconv.Atoi(bits)
+	checkConvertingError(err)
+
+	return rsabits
+}
+
+func getFullPath(path, attributeContent, regex string) string {
+	dir := filepath.Dir(path)
+	file := getRegexMatch(regex, attributeContent)
+
+	filePath := filepath.Join(dir, file)
+
+	return filePath
+}
+
+// AddCertificateInfo gets and adds certificate information of a certificate file
+func AddCertificateInfo(path, content string) map[string]interface{} {
+	var filePath string
+
+	if _, err := os.Stat(content); err != nil { // content is not a full valid path or is an incomplete path
+		filePath = getFullPath(path, content, `[0-9a-zA-Z-\/\\_]+\.pem`)
+	} else { // content is a full valid path
+		filePath = content
+	}
+
+	expirationDate := expirationDateFromPemCertificate(filePath)
+
+	if len(expirationDate) != 0 {
+		attributes := make(map[string]interface{})
+		attributes["file"] = filePath
+		attributes["expiration_date"] = expirationDate
+
+		rsaKeyBits := getBitRSAKeyBit(filePath)
+		if rsaKeyBits != -1 {
+			attributes["bit_rsa_key"] = rsaKeyBits
+		}
+		return attributes
+	}
+
+	return nil
+}
+
+func convert(m map[interface{}]interface{}) map[string]interface{} {
+	res := map[string]interface{}{}
+	for k, v := range m {
+		switch v2 := v.(type) {
+		case map[interface{}]interface{}:
+			res[fmt.Sprint(k)] = convert(v2)
+		default:
+			res[fmt.Sprint(k)] = v
+		}
+	}
+	return res
+}
+
+func readFile(filePath string) map[string]interface{} {
+	var result map[string]interface{}
+
+	content, err := os.ReadFile(filePath)
+	if err != nil {
+		log.Error().Msgf("Failed to read %s", filePath)
+	}
+
+	fileExtension := filepath.Ext(filePath)
+
+	if fileExtension == ".json" {
+		err := json.Unmarshal(content, &result)
+		if err != nil {
+			log.Error().Msgf("Failed to unmarshal '%s'", fileExtension)
+		}
+	} else if fileExtension == ".yaml" || fileExtension == ".yml" {
+		var resultYaml map[interface{}]interface{}
+		err := yaml.Unmarshal(content, &resultYaml)
+		if err != nil {
+			log.Error().Msgf("Failed to unmarshal '%s'", fileExtension)
+		}
+		result = convert(resultYaml)
+	}
+
+	return result
+}
+
+// AddSwaggerInfo gets and adds the content of a swagger file
+func AddSwaggerInfo(path, swaggerPath string) map[string]interface{} {
```

Comment:
Swagger related code should be placed under a different file from the certificate utils.